### ISSUE TITLE
[web] space key scrolls entire page, even if typing in textfield

### ISSUE THREAD

**Issue by devanflores** [flutter/flutter#74191]
## Steps to Reproduce
flutter run web  --web-renderer html

1. Run flutter web with html
2. Create a scrollable list (e.g. ListView and a textfield)
3. Type a spacebar in the textfield and the entire page will scroll down

**Expected results:** 
If a textfield is in focus, the page shouldn't scroll.
**Actual results:** 
The page scrolled when the spacebar was hit even though the textfield was in focus
<details>
  <summary>Logs</summary>

```
```

```
```

```
```
[√] Flutter (Channel dev, 1.26.0-8.0.pre, on Microsoft Windows [Version 10.0.19042.746], locale en-US)
    • Flutter version 1.26.0-8.0.pre at C:\Users\Devan\Downloads\flutter_windows_1.17.5-stable\flutter
    • Framework revision b9d06fffb2 (10 days ago), 2021-01-07 18:36:48 -0800
    • Engine revision 42a8d4c681
    • Dart version 2.12.0 (build 2.12.0-179.0.dev)

[X] Android toolchain - develop for Android devices
    X Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[√] VS Code (version 1.52.1)
    • VS Code at C:\Users\Devan\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.18.1

[√] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 87.0.4280.141
    • Edge (web)   • edge   • web-javascript • Microsoft Edge 87.0.664.75

</details>

**Comment by iapicca** [comment]
@devanflores 
I can reproduce the issue running `flutter run -d chrome --web-renderer html` with the latest master 

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(
      MaterialApp(
        home: Material(
          child: ListView(
            children: [
              ...[for (var i = 0; i < 5; ++i) const FlutterLogo(size: 80)],
              TextField(),
              ...[for (var i = 0; i < 20; ++i) const FlutterLogo(size: 80)],
            ],
          ),
        ),
      ),
    );

```

</details>

<details>
<summary>doctor</summary>

```console
[✓] Flutter (Channel master, 1.26.0-2.0.pre.402, on Linux, locale en_US.UTF-8)
    • Flutter version 1.26.0-2.0.pre.402 at /home/francesco/snap/flutter/common/flutter
    • Framework revision 2a188eeca3 (12 hours ago), 2021-01-17 19:27:00 -0800
    • Engine revision 609036f2bf
    • Dart version 2.12.0 (build 2.12.0-236.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/francesco/Android/Sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_SDK_ROOT = /home/francesco/Android/Sdk
    • Java binary at: /usr/lib/jvm/java-1.8.0-openjdk-amd64/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_275-8u275-b01-0ubuntu1~20.04-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /snap/bin/chromium

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Linux
    • Chrome (web)    • chrome • web-javascript • Chromium 87.0.4280.141 snap

! Doctor found issues in 1 category.

```
</details>

**Comment by TahaTesser** [comment]
Hi @devanflores 
Spacebar still scrolls when unfocused, tried different physics on listview, no help, or [this](https://stackoverflow.com/a/22559917)

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: ListView(
        children: [for (var i = 0; i < 30; i++) TextField()],
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () {},
      ),
    );
  }
}
```
</details>


<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel beta, 1.25.0-8.3.pre, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.25.0-8.3.pre at /Users/tahatesser/Code/flutter_beta
    • Framework revision 5d36f2e7f5 (4 days ago), 2021-01-14 15:57:49 -0800
    • Engine revision 7a8f8ca02c
    • Dart version 2.12.0 (build 2.12.0-133.7.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (2 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 87.0.4280.141

• No issues found!
```
```bash
[✓] Flutter (Channel dev, 1.26.0-8.0.pre, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.26.0-8.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision b9d06fffb2 (10 days ago), 2021-01-07 18:36:48 -0800
    • Engine revision 42a8d4c681
    • Dart version 2.12.0 (build 2.12.0-179.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • macOS (desktop)                    • macos         • darwin-x64     • macOS 11.1 20C69 darwin-x64
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 87.0.4280.141

• No issues found!
```
```bash
[✓] Flutter (Channel master, 1.26.0-2.0.pre.402, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.26.0-2.0.pre.402 at /Users/tahatesser/Code/flutter_master
    • Framework revision 2a188eeca3 (10 hours ago), 2021-01-17 19:27:00 -0800
    • Engine revision 609036f2bf
    • Dart version 2.12.0 (build 2.12.0-236.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • macOS (desktop)                    • macos         • darwin-x64     • macOS 11.1 20C69 darwin-x64
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 87.0.4280.141

• No issues found!
```

</details>

Thanks for filing the issue.

**Comment by mdebbar** [comment]
The 3 PRs necessary to fix this issue:
1. https://github.com/flutter/flutter/pull/74454 (already in the release candidate)
2. https://github.com/flutter/flutter/pull/74683 (already in the release candidate)
3. https://github.com/flutter/flutter/pull/74768 (not yet in the release candidate)

**Comment by amrahmed242** [comment]
after searching for any workaround available, replacing the space key default behavior with ActivateIntent (activates the currently focused control) is the only solution that worked for me

1.define the shortcuts overrides:
```
    final shortcuts = WidgetsApp.defaultShortcuts;
    shortcuts[LogicalKeySet(LogicalKeyboardKey.space)] = ActivateIntent();
```
2.place then globally in the material app
```
    return MaterialApp(
      shortcuts: shortcuts,
      // 
    );
```

**Comment by jonasbark** [comment]
I noticed it in the latest beta:
```
Flutter 2.1.0-12.2.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision 5bedb7b1d5 (vor 2 Wochen) • 2021-03-17 17:06:30 -0700
Engine • revision 711ab3fda0
Tools • Dart 2.13.0 (build 2.13.0-116.0.dev)
```

I actually had the issue that I could not use the space key when using an HTML Input Element:
```
 ui.platformViewRegistry.registerViewFactory(
      'gmap',
      (int viewId) {
        return TextInputElement()
          ..id = "autocomplete"
          ..style.outline = 'none'
          ..style.border = '0';
)}
...

 @override
  Widget build(BuildContext context) {
    return const SizedBox(
      height: 50,
      child: HtmlElementView(viewType: 'gmap'),
    );
  }
```

The workaround from @amrahmed242 also worked for me.

**Comment by diegoflassa** [comment]
Same issue here

`• Flutter version 2.2.2 at G:\users\Diego\Programming\SDKs\flutter`

    • Framework revision d79295af24 (3 weeks ago), 2021-06-11 08:56:01 -0700

    • Engine revision 91c9fc8fe0

    • Dart version 2.13.3`


The workaround from @amrahmed242 also worked for me too.

**Comment by Mcluky** [comment]
I know this issue is closed but I experienced the same problem on Flutter desktop. The workaround of @amrahmed242 didn't quite work for me but it got me an idea. In my case the spacebar shortcut is already set in the `WidgetsApp.defaultShortcuts`. For me the very `ActivateIntent()` was the problem. All I had to do to fix my issue was to remove the shortcut:

```dart
final shortcuts = WidgetsApp.defaultShortcuts;
shortcuts.remove(LogicalKeySet(LogicalKeyboardKey.space));

return MaterialApp.router(
   shortcuts: shortcuts,
   ...
)
```
